CI/macos/appcast_download.py

```diff
@@ -0,0 +1,62 @@
+import os
+import sys
+
+import requests
+import xmltodict
+
+
+def download_build(url):
+    print(f'Downloading build "{url}"...')
+    filename = url.rpartition("/")[2]
+    r = requests.get(url)
+    if r.status_code == 200:
+        with open(f"artifacts/{filename}", "wb") as f:
+            f.write(r.content)
+    else:
+        print(f"Build download failed, status code: {r.status_code}")
+        sys.exit(1)
+
+
+def read_appcast(url, fallback):
+    r = requests.get(url)
+    if r.status_code == 404:
+        print("Using fallback Appcast URL...")
+
+        r = requests.get(fallback)
+        if r.status_code != 200:
+            print(f"Appcast download failed, status code: {r.status_code}")
+            sys.exit(1)
+
+    filename = url.rpartition("/")[2]
+    with open(f"builds/{filename}", "wb") as f:
+        f.write(r.content)
+
+    appcast = xmltodict.parse(r.content, force_list=("item",))
+
+    dl = 0
+    for item in appcast["rss"]["channel"]["item"]:
+        channel = item.get("sparkle:channel", "stable")
+        if channel != target_branch:
+            continue
+
+        if dl == max_old_vers:
+            break
+        download_build(item["enclosure"]["@url"])
+        dl += 1
+
+
+if __name__ == "__main__":
+    appcast_urls = {
```

Comment:
Wouldn't it be better to use `plutil` to read the _actual_ appcast URLs from the app bundles that are downloaded in a prior step?

Reply:
We used to do it for notarisation:

https://github.com/obsproject/obs-studio/blob/6bee92d0545a73509c79af7ce5ea083e3d1e540f/CI/macos/03_package_obs.sh#L61

Reply:
The new version of this script mounts the DMG, reads the plist, and then uses the feed URL fromt here.

For this to correctly include existing releases when it is created for the first time we need to get R1CH or Jim to copy the existing appcasts to the new locations before tagging the first release with this workflow.